solutions/src/main/java/ru/sberbank/school/task02/Client.java

```diff
@@ -0,0 +1,134 @@
+        this.quotesService = quotesService;
+        this.conversionService = conversionService;
+        this.beneficiary = beneficiary;
+    }
+
+    public Client(ExternalQuotesService quotesService, FxConversionService conversionService) {
+        this(quotesService, conversionService, getBenificiary());
+    }
+
+    public static void main(String[] args) {
+        ServiceFactory serviceFactory = new ServiceFactoryImpl();
+        ExternalQuotesService externalQuotesService = new ExternalQuotesServiceDemo();
+        FxConversionService fxConversionService = serviceFactory.getFxConversionService(externalQuotesService);
+
+        Client client = new Client(externalQuotesService, fxConversionService);
+        FxRequest request = getFxRequestFromArgs(args);
+        FxResponse response = client.fetchResult(request);
+        System.out.println(request);
+        System.out.println(client.beneficiary);
+        System.out.println(response);
+    }
+
+    @Override
+    public List<FxResponse> fetchResult(List<FxRequest> requests) {
+        Objects.requireNonNull(requests);
+        List<FxResponse> resultList = new ArrayList<>();
+        for (FxRequest req : requests) {
+            resultList.add(fetchResult(req));
+        }
+        return resultList;
+    }
+
+    @Override
+    public FxResponse fetchResult(FxRequest requests) {
+        Objects.requireNonNull(requests);
+        BigDecimal amount = getAmount(requests);
+
+        BigDecimal price = conversionService.convert(getClientOperation(requests),
+                getSymbol(requests), amount);
+
+        return new FxResponse(requests.getSymbol(),
+                price.setScale(2, BigDecimal.ROUND_HALF_UP).toString(),
+                amount.setScale(2, BigDecimal.ROUND_HALF_UP).toString(),
+                LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")),
+                requests.getDirection(),
+                false);
+    }
+
+    private static FxRequest getFxRequestFromArgs(String[] args) {
+        String symbol;
+        String direction;
+        String amount;
+
+        if (args[0] == null) {
+            throw new NullPointerException("Wrong input argument.. Symbol is: NULL"
+                    + "expected \"USD/RUB\" or \"RUB/USD\"");
+        }
+        else if (args[0].toUpperCase().equals("USD/RUB")) {
+            symbol = "USD/RUB";
+        }else if (args[0].toUpperCase().equals("RUB/USD")) {
+            symbol = "RUB/USD";
+        }else {
+            throw new WrongSymbolException("Wrong input argument.. Symbol is: "
+                    + args[0] + " expected \"USD/RUB\" or \"RUB/USD\"");
+        }
+
+        if (args[1] == null) {
+            throw new NullPointerException("Wrong input argument.. Client operation is: NULL"
+                    + "expected \"BUY\" or \"SELL\"");
+        }else if (args[1].toUpperCase().equals("BUY")) {
+            direction = "BUY";
+        }else if (args[1].equals("SELL")) {
```

Comment:
WhitespaceAround: &#39;}&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1230317759>*